support/cas-server-support-trusted-mfa/src/main/java/org/apereo/cas/trusted/web/flow/DefaultDeviceFingerprintStrategy.java

```diff
@@ -0,0 +1,37 @@
+package org.apereo.cas.trusted.web.flow;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.springframework.core.OrderComparator;
+import org.springframework.webflow.execution.RequestContext;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Default {@link DeviceFingerprintStrategy} implementation that uses {@link DeviceFingerprintComponent} to generate
+ * a fingerprint.
+ *
+ * @author Daniel Frett
+ * @since 5.3.0
+ */
+@AllArgsConstructor
+public class DefaultDeviceFingerprintStrategy implements DeviceFingerprintStrategy {
+    @NonNull
+    private final List<DeviceFingerprintComponent> componentStrategies;
+    @NonNull
+    private final String componentSeparator;
+
+    @Override
+    public String determineFingerprint(@Nonnull final String principal, @Nonnull final RequestContext context,
```

Comment:
Please remove all `javax.annotation.Nonnull` from the code. These cause issues with JDK 9 and later in the future.

Reply:
👍 haven't done anything with JDK9 yet, but this is good to know.